### ISSUE TITLE
msys2 PyQtWebEngine

### ISSUE THREAD

**Issue by el07694** [msys2/MINGW-packages#8444]
I am trying to install PyQtWebEngine.
How can i do that?

Thanks in advance,
Chris Pappas

**Comment by ZachBacon** [comment]
Qtwebengine itself doesn't build on Mingw, so you won't be able to install it via msys2's mingw packages, at best you'd be able to install the Python bindings for qtwebkit. It's primarily a build issue for webengine, ergo the chromium engine to build on Mingw that I'm aware of.

**Comment by Biswa96** [comment]
Was there any previous attempt to build it?

**Comment by ZachBacon** [comment]
> Was there any previous attempt to build it?

I think so, but that was years ago, not so sure about now though.

**Comment by ZachBacon** [comment]
However I think it's hard coded build scripts still says mingw is not supported

**Comment by el07694** [comment]
https://packages.msys2.org/package/mingw-w64-x86_64-qtwebkit

that's all i need (<iframe> ajax chat)

**Comment by el07694** [comment]
Any updates? I want to  display a page with RTCPeerConnection (which is not available in QtWebKit).

**Comment by Alexpux** [comment]
@el07694 no one working on porting QWebEngine for Mingw

**Comment by MehdiChinoune** [comment]
We need nodejs to be able to build QtWebEngine (After some patches of course). Unfortunately nodejs is not buildable on MinGW-w64 for now.

**Comment by el07694** [comment]
Is there any alternative? (like sellenium, geckodriver) ?

**Comment by el07694** [comment]
From [qt wiki](https://wiki.qt.io/Building_Qt_5_from_Git)

The following packages are required to install Qt WebEngine:

sudo apt-get install libxcursor-dev libxcomposite-dev libxdamage-dev libxrandr-dev libxtst-dev libxss-dev libdbus-1-dev libevent-dev libfontconfig1-dev libcap-dev libpulse-dev libudev-dev libpci-dev libnss3-dev libasound2-dev libegl1-mesa-dev gperf bison nodejs

The libevent package is ready: https://packages.msys2.org/search?t=pkg&q=libevent .
The gperf package is ready: https://packages.msys2.org/search?t=pkg&q=gperf .
The bison package is ready: https://packages.msys2.org/search?t=pkg&q=bison .

All the other packages have to be prepared...

**Comment by Alexpux** [comment]
@el07694 stop spamming. Problem is not in dependencies (most deps that you write is Linux only). Problem in building QWebEngine itself.
You can try to do it yourself before posting here

**Comment by ZachBacon** [comment]
Yes and no, you only need the x development headers if you're compiling a native Linux app. Same with a few of the other dependencies. Currently the issue with qtwebengine is that right now it checks for a specific clang environment, and I mean one that's found in visual studio, not the mingw one. And that's just only the beginning.

**Comment by el07694** [comment]
I make an alternative choice  using selenium as  internal pyqt5 application.

Code:

`from PyQt5 import QtCore, QtGui, QtWidgets
import win32gui
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(800, 600)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.centralwidget)
        self.verticalLayout.setContentsMargins(0, 0, 0, 0)
        self.verticalLayout.setObjectName("verticalLayout")
        self.frame = QtWidgets.QFrame(self.centralwidget)
        self.frame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.frame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.frame.setObjectName("frame")
        #self.verticalLayout.addWidget(self.frame)
        MainWindow.setCentralWidget(self.centralwidget)
        
        self.chrome_options = Options()
        self.chrome_options.add_argument("--app=http://www.stackoverflow.com"); 
        self.s=Service(ChromeDriverManager().install())
        self.driver = webdriver.Chrome(service=self.s,options=self.chrome_options,service_log_path='NUL')
        #self.driver.get("http://www.stackoverflow.com")
        
        self.stack_overflow_hwnd = 0
        self.tries = 10
        self.total_tries = 0
        while(self.stack_overflow_hwnd==0 and self.total_tries<=self.tries):
            win32gui.EnumWindows(self.stack_overflow_hwnd_method, None)
            self.embed_window = QtGui.QWindow.fromWinId(self.stack_overflow_hwnd)
            self.embed_widget = QtWidgets.QWidget.createWindowContainer(self.embed_window)
            self.verticalLayout.addWidget(self.embed_widget)        
            self.tries+= 1
            time.sleep(1)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)
        
            
    def stack_overflow_hwnd_method(self, hwnd, ctx):
        window_title = win32gui.GetWindowText(hwnd)
        if "stack overflow" in window_title.lower():
            self.stack_overflow_hwnd = hwnd


    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())`

**Comment by ZachBacon** [comment]
And yet can't be provided as currently qtwebengine is only compilable on msvc not mingw-w64 yet

**Comment by Kreijstal** [comment]
> And yet can't be provided as currently qtwebengine is only compilable on msvc not mingw-w64 yet

noo propietary compiler :(

**Comment by el07694** [comment]
Anyway i am trying to switch off python/pyqt5 an use C++/Qt/Qtcreator/cmake instead

**Comment by realysy** [comment]
> @el07694 no one working on porting QWebEngine for Mingw

on progress: [https://github.com/msys2/MINGW-packages/pull/21914](https://github.com/msys2/MINGW-packages/pull/21914)
